### ISSUE TITLE
Error when trying to install on a Mura Site. 

### ISSUE THREAD

**Issue by Tigerlady** [gpearson/MuraEventRegistration#1]
Is this a publicly available plugin for Mura.  I tried downloading and installing it but got an error then started looking at the code and not sure it is meant to use as is.  

500 Error
Datasource: dsncbodev 

SQL: Show Tables LIKE 'eCaterers' 

Code: 0 

Type: database 

Incorrect syntax near the keyword 'LIKE'. 

Column: 0
ID: ??
Line: 24
Raw Trace: eventregistration.plugin.plugin_cfc$cf.udfCall(E:\Websites\dev.chamberbyohare.org\www\plugins\EventRegistration\plugin\plugin.cfc:24)
Template: E:\Websites\dev.chamberbyohare.org\www\plugins\EventRegistration\plugin\plugin.cfc
Type: cfml

**Comment by k1n6** [comment]
Could it be the type of database?

"show tables like xxx" seems like a mysql command to me.

**Comment by Tigerlady** [comment]
I am using MSSQL as the database.

**Comment by k1n6** [comment]
For fun I spun up a copy with MySql and it got further. But there were several errors in the install script and after several cycles of  Fix Error, uninstall, fix broken install, reinstall I gave up.

**Comment by Tigerlady** [comment]
Yeah I did the same thing.  definitely not ready for prime time. I have sort of came up with a different work around.

**Comment by gpearson** [comment]
I am in the process of making many changes since I have learned many new things about the MuraFW/1 Plugin. This was my first attempt at writing a plugin. This version needed to have code placed on the template of the site to work. 

For some reason I did not get any email notifications about this.

**Comment by k1n6** [comment]
Hi,

Yeah I am in need of a events registration module and was really interested in seeing what you have done here.  It seemed like some of the errors were typos and were easily fixed but other errors seemed more difficult for someone not familiar with the code base.

Are you going to update the github code? I'd be happy to give it another go and see how it goes.

I ended up temporarily deploying a fairly sparse events registration a co-worker had written quickly, but am going to have to rewrite / expand that or go with this - so I'd be willing to sort some things and make some contributions possibly.

Dan B.

**Comment by gpearson** [comment]
For some reason I am not getting updates to this issue when comments are posted. I have created a new branch called Development and going through all of the code to update it to the latest version of MuraFW/1 along with new things I have learned while creating my other 2 mura plugins that are needed here in Education. 

Is their a free version of MSSQL that can be installed in a VirtualBox environment that I can convert the Install sql files to? It has been over 10 years since I have messed with MSSQL. I am willing to port but it will take some learning on how to detect the version of the database server so the right sql scripts are ran.

**Comment by k1n6** [comment]
Yes mssql express is the version you are looking for

**Comment by gpearson** [comment]
Thank you. Been updating the Plugin all week and will continue this weekend. Just about completed the Upgrade to Version 3 which includes having all of the Bootstrap classes in place.

**Comment by k1n6** [comment]
Awesome, we used another little plugin for the display of a calendar, but I will be really hyped to see what you've done and maybe even contribute a little.

**Comment by gpearson** [comment]
I am having issues getting a virtual machine (VirtualBox) running with MSSQL Express and have Lucee/Railo connect to it so I can work out the MSSQL issues. I have MSSQL Installed but I have not found a good resource on how to get MSSQL Express to use a username/password so Railo/Lucee can connect to it so I can test the Install section of the plugin. At this point everything is under the Development Branch and I am just about done with all of the items which an Event Coordinator can do in the system.

**Comment by gpearson** [comment]
Now that I have a new version online for MySQL and learned all of the cfscript stuff to test for different databases. I am working on the code to create the MSSQL tables. Will spend time with MSSQL

**Comment by gpearson** [comment]
I have entered all of the tables for this plugin into MSSQL. I have exported an SQL Script. Any tricks on how to include the SQL Script into the install function of the MuraFW/1 Application so it can generate the necessary tables. Any Blog Entries or sample code I can look at.

**Comment by Tigerlady** [comment]
Good Morning,  Here is a sample script that I have used for some projects. I had to ad here as a text file just save it as a .cfm file and make what ever change you need to match your set up. 
[db-setup.txt](https://github.com/gpearson/MuraEventRegistration/files/434674/db-setup.txt)

**Comment by gpearson** [comment]
As of today, I have added scripts for MSSQL and have gone through each section of the application to resolve any database issues between mysql and mssql. All work has been done in the Development Branch of this project. 

This was a great learning experience for me as the development branch is a total rewrite of the application. Installation is much easier with this version and should work as expected. One note: after your create a new account you will need to manually update the database table tusersmemb to add a row for the UserID of Event Facilitator and the UserID of the newly created account. The Admin account in Mura is mapped to a future SiteAdmin subsystem that is not completed at this time.

**Comment by gpearson** [comment]
I have updated plugin install scripts to use Mura Function dbUtility to create tables,. columns, indexes and primary keys for all database structures. Plugin installs for those using MSSQL.
